### PR TITLE
Add elasticsearch

### PR DESCRIPTION
**Pull request by richtier**

### CODE REVIEW

File: spire/documents.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    updated = fields.DateField(attr="updated_datetime")
+
+    # not used, for parity with the lite-api application index. `Nested` and `Object` instead of NestedField and
+    # ObjectField so they are used only when generating index
+    queues = fields.Nested(doc_class=Queue)
+    submitted_by = fields.Object(doc_class=User)
+    case_officer = fields.Nested(doc_class=User)
+
+    class Index:
+        name = "spire-application-alias"
```
Comment (saruniitr) [inline spire/documents.py:215]: `ELASTICSEARCH_APPLICATION_INDEX_ALIAS?`

File: spire/models.py [hunk 2]
```diff
@@ -408,6 +408,21 @@ def __str__(self):
...
+        Application,
+        db_column='ela_id',
+        on_delete=models.DO_NOTHING,
+        related_name='application_case_officer_set'
+    )
+    case_officer = models.TextField()
+    working_days = models.IntegerField(blank=True, null=True)
+
+    class Meta:
+        managed = settings.SPIRE_DATABASE_MUTABLE
```
Comment (saruniitr) [inline spire/models.py:422]: If it is immutable then I guess we don't have to worry about updating index

File: example.env [hunk 3]
```diff
@@ -14,3 +14,5 @@ AWS_ACCESS_KEY_ID=debug
 AWS_SECRET_ACCESS_KEY=debug
 AWS_S3_REGION_NAME=debug
 S3_BUCKET_NAME=debug
+ELASTICSEARCH_HOST=http://localhost:9200
+FEATURE_ELASTICSEARCH_ON=False
```
Comment (saruniitr) [inline example.env:18]: missing newline

**Review by saruniitr — COMMENTED** [review]
few minor changes requested

File: spire/documents.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        queryset = super().get_value_from_instance(
+            instance, field_value_to_ignore=field_value_to_ignore
+        )
+        if queryset:
+            return resolve_value.get_value_from_instance(
+                queryset[0], field_value_to_ignore
+            )
+
+    def to_dict(self):
+        value = super().to_dict()
```
[the remark this answers is not in the document]
Comment (richtier) [reply spire/documents.py:86]: not as far as I can tell. it needs deleting as any kwarg passed to the field is then included in the dsl JSON - which sucks if we want to add custom attributes.

[the remark this answers is not in the document]
Comment (saruniitr) [reply spire/documents.py:32]: ok thanks, so if we don't delete then it is considered as search/filter attribute and can affect the results?

**Review by saruniitr — APPROVED** [review]
thanks for the changes, lgtm.

**Pull request merged**